### ISSUE TITLE
Validation should be tied to method

### ISSUE THREAD

**Issue by rebirthtobi** [proengsoft/laravel-jsvalidation#280]
There are cases where by i validate for different route methods in a single form request file using a switch statement, But JSValidation throws an error. 

Feature should be added to the package to accommodate this

**Comment by a-komarev** [comment]
Could you provide code example, please?

**Comment by rebirthtobi** [comment]
This is a sample code

```
public function rules()
    {
        switch($this->method()){
            case'POST':
                return [
                    'name' => 'required|string|max:255|unique:categories,name',
                    'description' => 'nullable|string|max:255',
                ];
            case 'PATCH':
                return [
                    'name' => [
                                'required',
                                'string',
                                'max:255',
                                Rule::unique('categories')->ignore($this->route('id')),
                            ],
                    'description' => 'nullable|string|max:255',
                ];
            default:
                break;
        }
    }
```

**Comment by a-komarev** [comment]
Similar issue was raised earlier: #255 
I don't see any way to implement it. I'm closing this issue since we don't know any way to solve it. Feel free to reopen it if you'll find the way.
